lib/checks/aria/unsupportedattr.js

```diff
@@ -1,16 +1,33 @@
-let unsupported = Array.from(node.attributes)
-	.filter(candidate => {
-		// filter out unsupported attributes
-		return axe.commons.aria.validateAttr(candidate.name, {
-			flagUnsupported: true
-		});
+const nodeName = node.nodeName.toUpperCase();
+const lookupTable = axe.commons.aria.lookupTable;
+const role = axe.commons.aria.getRole(node);
+
+const unsupportedAttrs = Array.from(node.attributes)
+	.filter(({ name }) => {
+		const attribute = lookupTable.attributes[name];
+		if (!axe.commons.aria.validateAttr(name) || !attribute.unsupported) {
+			return false;
+		}
+
+		const unsupported = attribute.unsupported;
+
+		if (typeof unsupported === 'boolean' || !unsupported.exceptions) {
+			return true;
```

Comment:
Nitpick, rather than checking if unsupported is falsey on line 8, I think this code would be a little easier to read if you check on line 14 if unsupported is not an object, and on line 14 do `return !!unsupported`.